Docs/Book/GettingStartedInPython.rst

```diff
@@ -3379,6 +3379,86 @@ These are adapted from the definitions in Gobbi, A. & Poppinger, D. “Genetic o
+
+Lipinski Rule of 5
+==================
+Lipinski's "Rule of 5" [#lipinski]_ was introduced to estimate the oral bioavailability of molecules. Poor absorption is likely if the molecule violates more than one of the following conditions: 
+
+* Molecular Weight <= 500 Da
+* No. Hydrogen Bond Donors <= 10
+* No. Hydrogen Bond Acceptors <= 5
+* LogP <= 5
+
+.. doctest::
+
+  >>> from rdkit import Chem
+  >>> from rdkit.Chem import Descriptors
+  >>> mol = Chem.MolFromSmiles('CC(=O)Nc1ccc(O)cc1')  # e.g. Paracetamol
+  >>> # Ro5 descriptors
+  >>> MW = Descriptors.MolWt(mol)
+  >>> HBA = Descriptors.NumHAcceptors(mol)
+  >>> HBD = Descriptors.NumHDonors(mol)
+  >>> LogP = Descriptors.MolLogP(mol)
+  >>> conditions = [MW <= 500, HBA <= 10, HBD <= 5, LogP <= 5]
+  >>> if conditions.count(True) >= 3:
+      ... pass_ro5 = True  # ro5 compliant
+  >>> else:
+      ... pass_ro5 = False  # fails ro5
+  >>> print(pass_ro5)
+
+Filtering Unwanted Substructures
+================================
+Pan Assay Interference Compounds (or PAINS) [#pains]_ are molecules that display non-specific binding, leading to unwanted side effects and false-positives in virtual screening. Common PAINS motifs include toxoflavin, isothiazolones, hydroxyphenyl hydrazones, curcumin, phenolsulfonamides, rhodanines, enones, quinones, and catechols. 
+
+The Brenk filter [#brenk]_ removes molecules containing substructures with undesirable pharmacokinetics or toxicity. These include sulfates and phosphates that contribute to unfavourable pharmacokinetics, nitro groups which are mutagenic and 2-halopyridines and thiols which are both reactive.
+
+The NIH filter [#jadhav]_, [#doveston]_ defined a list of functional groups with undesirable properties. These are split into those with reactive functionalities (including Michael acceptors, aldehydes, epoxides, alkyl halides, metals, 2-halo pyridines, phosphorus nitrogen bonds, α-chloroketones and β-lactams) and medicinal chemistry exclusions (including oximes, crown ethers, hydrazines, flavanoids, polyphenols, primary halide sulfates and multiple nitro groups).
+
+.. doctest::
+  >>> from rdkit import Chem
+  >>> from rdkit.Chem.FilterCatalog import FilterCatalog, FilterCatalogParams
+  
+  >>> mol = Chem.MolFromSmiles('CC1=C(C=C(C=C1)N2C(=O)C(=C(N2)C)N=NC3=CC=CC(=C3O)C4=CC(=CC=C4)C(=O)O)C')  # e.g. Eltrombopag
+  <BLANKLINE>
+  >>> # PAINS flag
+  >>> params_pains = FilterCatalogParams()
+  >>> params_pains.AddCatalog(FilterCatalogParams.FilterCatalogs.PAINS)
+  >>> catalog_pains = FilterCatalog(params_pains)
+  
+  >>> entry = catalog_pains.GetFirstMatch(mol)  # Get the first matching PAINS
+  >>> if entry is not None:
+      ... flag = True  # true if mol contains filter
+  >>> else:
+      ... flag = False  # false if not
```

Comment:
Since you aren't using the result that `GetFirstMatch()` returns, you can simplify this:
```suggestion
  >>> flag = catalog_pains.HasMatch(mol)  # Checks if there is a matching PAINS
```